<launch>
    <arg name="device_ip" default="192.168.1.200"/> <!-- 雷达ip -->
    <arg name="msop_port" default="2368" />         <!--雷达目的数据端口-->
    <arg name="difop_port" default="2369" />        <!--雷达目的设备端口-->
    <arg name="lidar_type" default="CX126S3"/>      <!--雷达型号: CX1S3 CX6S3 CH16S1 CH32A CH64W CB64S1_A CX126S3 CH128X1 CH128S1 CX128S2 CH256-->
    <arg name="use_time_service" default="false" /> <!--是否开启授时-->
    <arg name="packet_rate" default="9000.0"/>      <!-- PCAP文件回放速率，离线解析pcap包时使用 -->

  <node pkg="lslidar_driver" type="lslidar_driver_node" name="lslidar_driver_node" output="screen" ns="ch">
    <!--param name="pcap" value="$(find lslidar_driver)/pcap/xxx.pcap"/-->  <!--离线播pcap包路径-->
    <param name="packet_rate" value="$(arg packet_rate)"/>     <!--每秒读取pcap包数，离线解析pcap包时使用-->
    <param name="lidar_type" value="$(arg lidar_type)"/>
    <param name="device_ip" value="$(arg device_ip)"/>
    <param name="msop_port" value="$(arg msop_port)" />
    <param name="difop_port" value="$(arg difop_port)"/>
    <param name="use_time_service" value="$(arg use_time_service)"/>
    <param name="add_multicast" value="false"/>     <!--雷达是否启用组播模式-->
    <param name="group_ip" value="224.1.1.2"/>      <!--组播IP-->
    <param name="frame_id" value="laser_link"/>     <!--雷达点云坐标系ID-->
    <Param name="pointcloud_topic" value="lslidar_point_cloud"/>  <!--点云话题名称-->
    <param name="min_range" value="0.3"/>           <!--最小距离 小于此值的点将被过滤 可根据实际需求更改 单位:米-->
    <param name="max_range" value="500.0"/>         <!--最大距离 大于此值的点将被过滤 可根据实际需求更改 单位:米-->
    <param name="scan_start_angle" value="3000"/>   <!--最小角度 小于此值的点将被过滤 可根据实际需求更改 单位:0.01°-->
    <param name="scan_end_angle" value="15000"/>    <!--最大角度 大于此值的点将被过滤 可根据实际需求更改 单位:0.01°-->
    <param name="horizontal_angle_resolution" value="0.15"/> <!--laserscan角度分辨率,根据雷达型号修改-->
    <param name="publish_scan" value="false"/> <!--是否发布laserscan-->
    <param name="scan_num" value="64"/>             <!--laserscan线号-->
    <param name="echo_mode" value="0"/> <!--双回波模式下生效 0:发布全部点云  1:发布第一次回波点云  2:发布第二次回波点云-->
  
    <param name="is_pretreatment" value="false"/> <!--是否开启预处理 仅对点云生效-->
    <param name="x_offset" value="0.0"/>  <!-- rviz 中表现为围绕红色的 x 轴偏移量 单位: m-->
    <param name="y_offset" value="0.0"/>  <!-- rviz 中表现为围绕绿色的 y 轴偏移量 单位: m-->
    <param name="z_offset" value="0.0"/>  <!-- rviz 中表现为围绕蓝色的 z 轴偏移量 单位: m-->
    <param name="roll" value="0.0"/>      <!-- rviz 中表现为围绕红色的 x 轴旋转   单位: rad-->
    <param name="pitch" value="0.0"/>     <!-- rviz 中表现为围绕绿色的 y 轴旋转   单位: rad-->
    <param name="yaw" value="0.0"/>       <!-- rviz 中表现为围绕蓝色的 z 轴旋转   单位: rad-->
  </node>

  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find lslidar_driver)/rviz/lslidar_ch.rviz"/> <!-- 注释此行不启动RViz -->
</launch>
